### ISSUE TITLE
Fails to build with Boost 1.72.0

### ISSUE THREAD

**Issue by jbeich** [rstudio/rstudio#5742]
Regressed by https://github.com/boostorg/filesystem/commit/9a14c37d6f95 See also: [rstudio 1.2.5019 build log](https://github.com/rstudio/rstudio/files/3847996/RStudio-1.2.5019.log), [boost 1.72 source](https://dl.bintray.com/boostorg/master/boost_1_72_0-snapshot.tar.bz2) ([downstream](https://reviews.freebsd.org/D22136)).
CC @yurivict as it may break [FreeBSD package](https://www.freshports.org/devel/RStudio) after [2019-12-11](https://www.boost.org/development/) (Boost 1.72 release).
```
$ c++ --version
FreeBSD clang version 8.0.1 (tags/RELEASE_801/final 366581) (based on LLVM 8.0.1)
Target: x86_64-unknown-freebsd12.1
Thread model: posix
InstalledDir: /usr/bin
$ git describe --tags --always
v1.2.5019-1804-g1330290022
```
```c++
$ cmake ../rstudio
$ make
[  0%] Building CXX object src/cpp/shared_core/CMakeFiles/rstudio-shared-core.dir/FilePath.cpp.o
cd /path/to/rstudio_build/src/cpp/shared_core && /usr/bin/c++  -DBOOST_ASIO_DISABLE_KQUEUE -DBOOST_ENABLE_ASSERT_HANDLER -DBOOST_SIGNALS_NO_DEPRECATION_WARNING -DRSTUDIO_BOOST_NAMESPACE=boost -DRSTUDIO_BOOST_SIGNALS_VERSION=2 -DRSTUDIO_UNIT_TESTS_ENABLED -DWEBSOCKETPP_STRICT_MASKING -D_FORTIFY_SOURCE=2 -D_GLIBCXX_ASSERTIONS -D_WEBSOCKETPP_NO_CPP11_MEMORY_=1 -DSTRIPPED_FILENAME=\"src/cpp/shared_core/FilePath.cpp\" -I/path/to/rstudio/src/cpp/shared_core/include -isystem /usr/local/include -isystem /opt/rstudio-tools/crashpad -isystem /opt/rstudio-tools/crashpad/crashpad -isystem /opt/rstudio-tools/crashpad/crashpad/third_party/mini_chromium/mini_chromium  -std=c++0x -std=c++11 -g -fPIC   -Wall -pthread -Wformat -Wformat-security -fstack-protector --param ssp-buffer-size=4 -fPIC -O0 -o CMakeFiles/rstudio-shared-core.dir/FilePath.cpp.o -c /path/to/rstudio/src/cpp/shared_core/FilePath.cpp
/path/to/rstudio/src/cpp/shared_core/FilePath.cpp:756:40: error: no member named 'level' in
      'boost::filesystem::recursive_directory_iterator'
         if (!in_iterationFunction(itr.level(),
                                   ~~~ ^
1 error generated.
```
According to Boost.Filesystem [release notes](https://github.com/boostorg/filesystem/blob/master/doc/release_history.html):
> * **Deprecated:** The following members of `recursive_directory_iterator` are now marked as deprecated: `level()`, `no_push_pending()`, `no_push_request()`, `no_push()`. Users are advised to replace their use with the standard counterparts: `depth()`, `recursion_pending()`, `disable_recursion_pending()`. Note that `recursion_pending()` has the opposite meaning compared to `no_push_pending()` and `no_push_request()`. Deprecated methods will be removed in a future release.

**Comment by kevinushey** [comment]
Or just don't define `BOOST_FILESYSTEM_NO_DEPRECATED` for now.

**Comment by jbeich** [comment]
I confirm, builds fine without the following line.
https://github.com/rstudio/rstudio/blob/ca3fb055862c8d422a0f20b053a322134fc590b8/src/cpp/shared_core/FilePath.cpp#L39

**Comment by kevinushey** [comment]
Fixed on master; I've also backported the fix to the v1.2 branch and it will become part of a later patch release.
